controllers/drplacementcontrol_controller.go

```diff
@@ -1759,3 +1735,56 @@ func selectVRGNamespace(
 		return drpc.Namespace, nil
 	}
 }
+
+func addOrUpdateCondition(conditions *[]metav1.Condition, conditionType string,
+	observedGeneration int64, status metav1.ConditionStatus, reason, msg string,
+) bool {
+	newCondition := metav1.Condition{
+		Type:               conditionType,
+		Status:             status,
+		ObservedGeneration: observedGeneration,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            msg,
+	}
+
+	existingCondition := findCondition(*conditions, conditionType)
+	if existingCondition == nil ||
+		existingCondition.Status != newCondition.Status ||
+		existingCondition.ObservedGeneration != newCondition.ObservedGeneration ||
+		existingCondition.Reason != newCondition.Reason ||
+		existingCondition.Message != newCondition.Message {
+		setStatusCondition(conditions, newCondition)
+
+		return true
+	}
+
+	return false
+}
+
+// Initial creation of the DRPC status condition. This will also preserve the ordering of conditions in the array
+func ensureDRPCConditionsInited(conditions *[]metav1.Condition, observedGeneration int64, message string) {
+	const DRPCTotalConditions = 2
+	if len(*conditions) == DRPCTotalConditions {
+		return
+	}
+
+	time := metav1.NewTime(time.Now())
+
+	setStatusConditionIfNotFound(conditions, metav1.Condition{
+		Type:               rmn.ConditionAvailable,
+		Reason:             string(rmn.Initiating),
+		ObservedGeneration: observedGeneration,
+		Status:             metav1.ConditionTrue,
```

Comment:
Can we start with Unknown as the initial condition status, true always gives me the jitters in other parts of the code where we compare at times the status and the generation.

Reply:
I like to start with the status set to Unknown as well. But during testing, there was a case where that will break. I don't remember that one. In any case, True won't hurt. Especially, on the initial run of the DRPC instance, the world is good until it isn't.

Reply:
We can add a comment to consider starting with Unknown state. Starting with fake "good" state causes lot of trouble in other case, like restarting minikube env. We get fake status for about 30 seconds from everything and this is very hard to manage. It may not be relevant to drpc.

Reply:
Again, start positive, until there is a need to switch.  Starting with `Unknown` will just cause more trouble for other clients. The UI is one of them.